File: LinguaRise/LinguaRise.Models/DTOs/Word/WordDTO.cs
namespace LinguaRise.Models.DTOs;

public class WordDTO
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Level { get; set; }
    public int LanguageId { get; set; }
    public string LanguageCode { get; set; }
    public int VocabularyCategoryId { get; set; }
    public string VocabularyCategoryName { get; set; }
}

File: LinguaRise/linguarise.app/src/data/WordDTO.js
export class WordDTO {
  constructor(data = {}) {
    this.id = data.Id ?? data.id;
    this.name = data.Name ?? data.name;

    const levelData = data.Level ?? data.level;
    this.level =
      levelData && typeof levelData === "object"
        ? levelData.Value ?? levelData.value ?? null
        : levelData ?? null;

    this.vocabularyCategoryName =
      data.VocabularyCategoryName ?? data.vocabularyCategoryName;
  }
}